dconf fails tests on armv7l-linux

Building `dconf` fails on `armv7l-linux`. The shared memory test fails.

The complete log is below:

```
building '/nix/store/5yqhikpp5sjl422s28vg71fcjym4q5b9-dconf-0.30.1.drv'...
unpacking sources
unpacking source archive /nix/store/canh30a3lr5pmqdr6jgnn01qh4qzyhm2-dconf-0.30.1.tar.xz
source root is dconf-0.30.1
setting SOURCE_DATE_EPOCH to timestamp 1540245184 of file dconf-0.30.1/trim-lcov.py
patching sources
applying patch /nix/store/a63acqqz1bjda6fal37gy0n8jsjjsh53-49f4d916e1151af3975df52c522c69de98ed2fbb.patch
patching file gvdb/gvdb-reader.c
patching file gvdb/gvdb-reader.h
patching script interpreter paths in meson_post_install.py
meson_post_install.py: interpreter directive changed from "/usr/bin/env python3" to "/nix/store/ihxv5wfgg5dri0q2aijzvp6wgp56m1j9-python3-3.7.2/bin/python3"
updateAutotoolsGnuConfigScriptsPhase
configuring
meson flags: --buildtype=release         --libdir=/nix/store/w26gknp61y08q7jfs4xf3gr9p0mk19a3-dconf-0.30.1-lib/lib --libexecdir=/nix/store/w26gknp61y08q7jfs4xf3gr9p0mk19a3-dconf-0.30.1-lib/libexec         --bindir=/nix/store/nr3b9r88fq4b22i1zirgj01mnhggq3ny-dconf-0.30.1/bin --sbindir=/nix/store/nr3b9r88fq4b22i1zirgj01mnhggq3ny-dconf-0.30.1/sbin         --includedir=/nix/store/47klh3xngccfbmwvwq80c9rsvjr3asa7-dconf-0.30.1-dev/include         --mandir=/nix/store/nr3b9r88fq4b22i1zirgj01mnhggq3ny-dconf-0.30.1/share/man --infodir=/nix/store/nr3b9r88fq4b22i1zirgj01mnhggq3ny-dconf-0.30.1/share/info         --localedir=/nix/store/w26gknp61y08q7jfs4xf3gr9p0mk19a3-dconf-0.30.1-lib/share/locale         -Dauto_features=disabled         --prefix=/nix/store/nr3b9r88fq4b22i1zirgj01mnhggq3ny-dconf-0.30.1 --sysconfdir=/etc -Dgtk_doc=true
The Meson build system
Version: 0.49.1
Source dir: /build/dconf-0.30.1
Build dir: /build/dconf-0.30.1/build
Build type: native build
Project name: dconf
Project version: 0.30.1
Native C compiler: /nix/store/yf357wms27m8z59zf6n6jbld34kfa560-gcc-wrapper-7.4.0/bin/cc (gcc 7.4.0 "gcc (GCC) 7.4.0")
Native Vala compiler: valac (valac 0.42.4)
Build machine cpu family: arm
Build machine cpu: armv7l
Found pkg-config: /nix/store/ksp2wzaa6yrzxcwqdcvz1s8c3afai9iy-pkg-config-0.29.2/bin/pkg-config (0.29.2)
Dependency gio-2.0 found: YES 2.58.2
Dependency gio-unix-2.0 found: YES 2.58.2
Dependency glib-2.0 found: YES 2.58.2
Dependency dbus-1 found: YES 1.12.12
Dependency vapigen found: YES 0.42.4
Dependency bash-completion found: YES 2.8
Configuring config.h using configuration
Compiler for C supports arguments -fvisibility=hidden: YES
Configuring ca.desrt.dconf.service using configuration
Program abicheck.sh found: YES (/build/dconf-0.30.1/gsettings/abicheck.sh)
Library posix found: YES
Program xsltproc found: YES (/nix/store/mi0368ff20z9ji5467klm24rp2jaqnb5-libxslt-1.1.33-bin/bin/xsltproc)
Library dl found: YES
Library m found: YES
Program meson_post_install.py found: YES (/build/dconf-0.30.1/meson_post_install.py)
Build targets in project: 29
Found ninja-1.9.0 at /nix/store/3ixxaybak98v52cc29r9z5miswdbncw4-ninja-1.9.0/bin/ninja
meson: enabled parallel building
building
build flags: -j1 -l1
[1/77] Compiling C object 'shm/feecd04@@dconf-shm@sta/dconf-shm.c.o'.
[2/77] Linking static target shm/libdconf-shm.a.
[3/77] Compiling C object 'gvdb/6f5c6fc@@gvdb@sta/gvdb-builder.c.o'.
[4/77] Compiling C object 'gvdb/6f5c6fc@@gvdb@sta/gvdb-reader.c.o'.
[5/77] Linking static target gvdb/libgvdb.a.
[6/77] Compiling C object 'common/92a5dc0@@dconf-common@sta/dconf-changeset.c.o'.
[7/77] Compiling C object 'common/92a5dc0@@dconf-common@sta/dconf-error.c.o'.
[8/77] Compiling C object 'common/92a5dc0@@dconf-common@sta/dconf-paths.c.o'.
[9/77] Linking static target common/libdconf-common.a.
[10/77] Compiling C object 'common/92a5dc0@@dconf-common-hidden@sta/dconf-changeset.c.o'.
[11/77] Compiling C object 'common/92a5dc0@@dconf-common-hidden@sta/dconf-error.c.o'.
[12/77] Compiling C object 'common/92a5dc0@@dconf-common-hidden@sta/dconf-paths.c.o'.
[13/77] Compiling C object 'engine/ed9f6f2@@dconf-engine@sta/dconf-engine.c.o'.
[14/77] Linking static target common/libdconf-common-hidden.a.
[15/77] Compiling C object 'engine/ed9f6f2@@dconf-engine@sta/dconf-engine-profile.c.o'.
[16/77] Compiling C object 'engine/ed9f6f2@@dconf-engine@sta/dconf-engine-source.c.o'.
[17/77] Compiling C object 'engine/ed9f6f2@@dconf-engine@sta/dconf-engine-source-file.c.o'.
[18/77] Compiling C object 'engine/ed9f6f2@@dconf-engine@sta/dconf-engine-source-user.c.o'.
[19/77] Compiling C object 'engine/ed9f6f2@@dconf-engine@sta/dconf-engine-source-service.c.o'.
[20/77] Compiling C object 'engine/ed9f6f2@@dconf-engine@sta/dconf-engine-source-system.c.o'.
[21/77] Generating dconf-generated.c with a custom command.
[22/77] Linking static target engine/libdconf-engine.a.
[23/77] Generating dconf-generated.h with a custom command.
[24/77] Compiling C object 'service/9df6b02@@dconf-service@sta/meson-generated_.._dconf-generated.c.o'.
[25/77] Compiling C object 'service/9df6b02@@dconf-service@sta/dconf-blame.c.o'.
[26/77] Compiling C object 'service/9df6b02@@dconf-service@sta/dconf-gvdb-utils.c.o'.
[27/77] Compiling C object 'service/9df6b02@@dconf-service@sta/dconf-shm-writer.c.o'.
[28/77] Compiling C object 'service/9df6b02@@dconf-service@sta/dconf-writer.c.o'.
../service/dconf-writer.c: In function 'dconf_writer_real_commit':
../service/dconf-writer.c:178:7: warning: ignoring return value of 'write', declared with attribute warn_unused_result [-Wunused-result]
       write (invalidate_fd, "\0\0\0\0\0\0\0\0", 8);
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[29/77] Compiling C object 'service/9df6b02@@dconf-service@sta/dconf-keyfile-writer.c.o'.
[30/77] Compiling C object 'service/9df6b02@@dconf-service@sta/dconf-service.c.o'.
[31/77] Linking static target service/libdconf-service.a.
[32/77] Compiling C object 'service/9df6b02@@dconf-service@exe/main.c.o'.
[33/77] Linking target service/dconf-service.
[34/77] Compiling C object 'gdbus/8d60afc@@dconf-gdbus-thread@sta/dconf-gdbus-thread.c.o'.
[35/77] Linking static target gdbus/libdconf-gdbus-thread.a.
[36/77] Compiling C object 'gdbus/8d60afc@@dconf-gdbus-filter@sta/dconf-gdbus-filter.c.o'.
[37/77] Linking static target gdbus/libdconf-gdbus-filter.a.
[38/77] Compiling C object 'gsettings/df7c3da@@dconfsettings@sha/dconfsettingsbackend.c.o'.
[39/77] Compiling C object 'client/948fe60@@dconf-client@sta/dconf-client.c.o'.
[40/77] Linking target gsettings/libdconfsettings.so.
[41/77] Linking static target client/libdconf-client.a.
[42/77] Compiling C object 'client/948fe60@@dconf@sha/dconf-client.c.o'.
[43/77] Compiling C object 'bin/51a1f05@@dconf@exe/.._gvdb_gvdb-builder.c.o'.
[44/77] Linking target client/libdconf.so.1.0.0.
[45/77] Generating symbol file 'client/948fe60@@dconf@sha/libdconf.so.1.0.0.symbols'.
[46/77] Compiling Vala source ../client/dconf.vapi ../bin/gvdb.vapi ../bin/dconf-dump.vala ../bin/dconf-update.vala ../bin/dconf.vala.
[47/77] Compiling C object 'bin/51a1f05@@dconf@exe/meson-generated_dconf-dump.c.o'.
[48/77] Compiling C object 'bin/51a1f05@@dconf@exe/meson-generated_dconf-update.c.o'.
[49/77] Compiling C object 'bin/51a1f05@@dconf@exe/meson-generated_dconf.c.o'.
[50/77] Linking target bin/dconf.
[51/77] Generating dconf.1 with a custom command.
[52/77] Compiling C object 'tests/59830eb@@dconf-mock@sta/dconf-mock-dbus.c.o'.
[53/77] Generating dconf.7 with a custom command.
[54/77] Generating dconf-service.1 with a custom command.
[55/77] Compiling C object 'tests/59830eb@@dconf-mock@sta/dconf-mock-gvdb.c.o'.
[56/77] Compiling C object 'tests/59830eb@@dconf-mock@sta/dconf-mock-shm.c.o'.
[57/77] Linking static target tests/libdconf-mock.a.
[58/77] Compiling C object 'tests/59830eb@@paths@exe/paths.c.o'.
[59/77] Linking target tests/paths.
[60/77] Compiling C object 'tests/59830eb@@changeset@exe/changeset.c.o'.
[61/77] Linking target tests/changeset.
[62/77] Compiling C object 'tests/59830eb@@shm@exe/shm.c.o'.
[63/77] Compiling C object 'tests/59830eb@@shm@exe/tmpdir.c.o'.
[64/77] Compiling C object 'tests/59830eb@@gvdb@exe/gvdb.c.o'.
[65/77] Linking target tests/shm.
[66/77] Linking target tests/gvdb.
[67/77] Compiling C object 'tests/59830eb@@gdbus-thread@exe/dbus.c.o'.
[68/77] Compiling C object 'tests/59830eb@@gdbus-filter@exe/dbus.c.o'.
[69/77] Linking target tests/gdbus-thread.
[70/77] Linking target tests/gdbus-filter.
[71/77] Compiling C object 'tests/59830eb@@engine@exe/engine.c.o'.
[72/77] Linking target tests/engine.
[73/77] Compiling C object 'tests/59830eb@@client@exe/client.c.o'.
[74/77] Linking target tests/client.
[75/77] Compiling C object 'tests/59830eb@@writer@exe/meson-generated_.._.._service_dconf-generated.c.o'.
[76/77] Compiling C object 'tests/59830eb@@writer@exe/writer.c.o'.
[77/77] Linking target tests/writer.
running tests
check flags: -j1 -l1 test
[0/1] Running all tests.
 1/10 abicheck                                OK       0.05 s
 2/10 paths                                   OK       0.02 s
 3/10 changeset                               OK       0.34 s
 4/10 shm                                     FAIL     0.04 s (killed by signal 5 SIGTRAP)
 5/10 gvdb                                    OK      15.31 s
 6/10 gdbus-thread                            SKIP     0.04 s
 7/10 gdbus-filter                            SKIP     0.04 s
 8/10 engine                                  OK       6.15 s
 9/10 client                                  OK       0.04 s
10/10 writer                                  OK       0.04 s

Ok:                    7
Expected Fail:         0
Fail:                  1
Unexpected Pass:       0
Skipped:               2
Timeout:               0


The output from the failed tests:

 4/10 shm                                     FAIL     0.04 s (killed by signal 5 SIGTRAP)

--- command ---
G_TEST_BUILDDIR='/build/dconf-0.30.1/build/tests' G_DEBUG='gc-friendly,fatal-warnings' G_TEST_SRCDIR='/build/dconf-0.30.1/tests' MALLOC_CHECK_='2' LC_ALL='C.UTF-8' /build/dconf-0.30.1/build/tests/shm
--- stdout ---
/shm/mkdir-fail: OK
/shm/close-null: OK
/shm/open-and-flag:
--- stderr ---

(/build/dconf-0.30.1/build/tests/shm:2824): dconf-CRITICAL **: 09:59:39.862: failed to allocate file '/build/dconf-testcase.Q55F0Z/dconf/foo': Invalid argument.  dconf will not work properly.
-------

Full log written to /build/dconf-0.30.1/build/meson-logs/testlog.txt
FAILED: meson-test
/nix/store/5kn3lj7mimz6vl6ihd1pl7v5y1p3pj1l-meson-0.49.1/bin/meson test --no-rebuild --print-errorlogs
ninja: build stopped: subcommand failed.
builder for '/nix/store/5yqhikpp5sjl422s28vg71fcjym4q5b9-dconf-0.30.1.drv' failed with exit code ```

--- a/pkgs/desktops/gnome-3/core/dconf/default.nix
+++ b/pkgs/desktops/gnome-3/core/dconf/default.nix
@@ -38,7 +38,7 @@ stdenv.mkDerivation rec {
     "-Dgtk_doc=true"
   ];
 
-  doCheck = !stdenv.isAarch64 && !stdenv.isDarwin;
+  doCheck = !stdenv.isAarch32 && !stdenv.isAarch64 && !stdenv.isDarwin;
 
   passthru = {
     updateScript = gnome3.updateScript {

